core/routemgmt/deleteApi.js

```diff
@@ -0,0 +1,448 @@
+              console.log('API configuration '+updatedDbDocOrErrMsg._id+' successfully updated');
+              return Promise.resolve();
+          })
+      } else {
+          if (dbdoc.gwApiActivated && !force) {
+              return Promise.reject('API is active and cannot be deleted.  Once the API is deactivated, it can be deleted.')
+          }
+          return deleteGatewayApi(gwInfo, dbdoc.gwApiGuid)
+          .then(function() {
+              console.log('Gateway API '+dbdoc.gwApiGuid+' successfully removed from gateway');
+              return deleteApiFromDb(cloudantDb, dbdoc._id, dbdoc._rev)
+              .then(function() {
+                  console.log('API configuration '+dbdoc._id+' successfully removed from the database');
+                  return Promise.resolve();
+              })
+          })
+      }
+  })
+  .catch(function(reason) {
+      // FIXME MWD Possibly need to rollback some operations ??
+      console.error('API deletion failure: '+reason);
+      return Promise.reject('API deletion failure: '+reason);
+  });
+}
+
+
+function getCloudantAccount(message) {
+  // full cloudant URL - Cloudant NPM package has issues creating valid URLs
+  // when the username contains dashes (common in Bluemix scenarios)
+  var cloudantUrl;
+
+  if (message.url) {
+    // use bluemix binding
+    cloudantUrl = message.url;
+  } else {
+    if (!message.host) {
+      whisk.error('cloudant account host is required.');
+      return;
+    }
+    if (!message.username) {
+      whisk.error('cloudant account username is required.');
+      return;
+    }
+    if (!message.password) {
+      whisk.error('cloudant account password is required.');
+      return;
+    }
+    if (!message.port) {
+      whisk.error('cloudant account port is required.');
+      return;
+    }
+    if (!message.protocol) {
+      whisk.error('cloudant account protocol is required.');
+      return;
+    }
+
+    cloudantUrl = message.protocol + "://" + message.username + ":" + message.password + "@" + message.host + ":" + message.port;
+  }
+
+  return require('cloudant')({
+    url: cloudantUrl
+  });
+}
+
+// Retrieve the specific API document from the DB
+// This is a wrapper around an action invocation (getApi)
+function getDbApiDoc(namespace, basepath) {
+  var actionName = '/whisk.system/routemgmt/getApi';
+  var params = {
+    'namespace': namespace,
+    'basepath': basepath
+  }
+  console.log('getDbApiDoc() for namespace:basepath: '+namespace+':'+basepath);
+  return whisk.invoke({
+    name: actionName,
+    blocking: true,
+    parameters: params
+  })
+  .then(function (activation) {
+    console.log('whisk.invoke('+actionName+', '+params.namespace+', '+params.basepath+') ok');
+    console.log('Results: '+JSON.stringify(activation));
+    if (activation && activation.result && activation.result.apis &&
+        activation.result.apis.length == 1 && activation.result.apis[0].value &&
+        activation.result.apis[0].value._rev) {
```

Comment:
I find that defensive coding quite annoying to read, while i realize its necessary.

Will open an issue to add lodash to our runtimes. You could use https://lodash.com/docs/4.16.4#has then, which would make that way more readable.